File: java/1_Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numsSwap = new HashMap<Integer, Integer>();
        int[] result = {-1, -1};
        for (int i = 0; i < nums.length; ++i) {
            if (numsSwap.containsKey(target - nums[i])) {
                result[0] = i;
                result[1] = numsSwap.get(target - nums[i]);
            } else {
                numsSwap.put(nums[i], i);
            }
        }

        return result;
    }
}

File: go/1_Two_Sum.go
func twoSum(nums []int, target int) []int {
    numsSwap := make(map[int]int)
    var result []int
    for i:= 0; i < len(nums); i++ {
        if numsSwap[target - nums[i]] != 0 {
            result = append(result, numsSwap[target - nums[i]] - 1)
	        result = append(result, i)
        } else {
            numsSwap[nums[i]] = i + 1
        }
    }
    
    return result;
}

